### PR TITLE
WRQ-19324: Remove findDOMNode in Dropdown and ContextualPopupDecorator

### PR DESCRIPTION
**Pull request by juwonjeong**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
As `findDOMNode` will be removed in React 19, replace/remove `findDOMNode` in sandstone code.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Remove `findDOMNode` in contextualPopupDecorator and added a sibling DOM node to the Wrapped.

Remove `findDOMNode` in DropdownList and added a sibling DOM node to the Wrapped to find dropdownListNode in handleFocus function. 
And configuring the dropdownList node's enterTo value as default-element is removed. Because the logic(https://github.com/enactjs/sandstone/commit/4249b22035e957d1fe0f70cb297f5c23511ff195) already exists that spot the selected item without `enterTo` configuration.  

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRQ-19324

### CODE REVIEW

**Comment by MikyungKim** [comment]
As we discussed, let's add some changelog for this.

**Review by MikyungKim — APPROVED** [review]
LGTM
I think it's okay for `ContextualPopupDecorator` as is since it's not a hoc that can be applied many times.
I confirmed `Dropdown` list is working as expected without the removed code when it has given `open` and `selected` prop.

File: ContextualPopupDecorator/ContextualPopupDecorator.js [hunk 1]
```diff
@@ -274,7 +273,7 @@ const Decorator = hoc(defaultConfig, (config, Wrapped) => {
 			this.overflow = {};
 			this.adjustedDirection = this.props.direction;
 			this.id = this.generateId();
-
+			this.clientSiblingRef = createRef(null);
```
Comment (MikyungKim) [inline ContextualPopupDecorator/ContextualPopupDecorator.js:276]:
```suggestion
			this.clientSiblingRef = createRef(null);
			
```

Comment (juwonjeong) [reply ContextualPopupDecorator/ContextualPopupDecorator.js:276]: It's fixed:)

**Review by 0x64 — APPROVED** [review]

**Pull request merged**